src/main/java/com/watch/shop/app/controller/MenuHandler.java

```diff
@@ -0,0 +1,143 @@
+import com.watch.shop.app.model.entity.OperationSystem;
+import com.watch.shop.app.model.entity.QuartzWatch;
+import com.watch.shop.app.model.entity.QuartzWatch.QuartzWatchBuilder;
+import com.watch.shop.app.model.entity.SmartWatch;
+import com.watch.shop.app.model.entity.SmartWatch.SmartWatchBuilder;
+import com.watch.shop.app.model.entity.SolarWatch;
+import com.watch.shop.app.model.entity.SolarWatch.SolarWatchBuilder;
+import com.watch.shop.app.model.entity.Watch;
+import com.watch.shop.app.model.entity.Watch.WatchBuilder;
+import com.watch.shop.app.model.entity.WatchType;
+import com.watch.shop.app.view.WatchView;
+import lombok.RequiredArgsConstructor;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.List;
+
+import static com.watch.shop.app.controller.MenuConstants.END_SEPARATOR;
+import static com.watch.shop.app.controller.MenuConstants.FORMAT_HEADER;
+import static com.watch.shop.app.controller.MenuConstants.FORMAT_ROW;
+import static com.watch.shop.app.controller.MenuConstants.FORMAT_SEPARATOR;
+import static com.watch.shop.app.controller.MenuConstants.MENU;
+import static com.watch.shop.app.controller.MenuConstants.SORT_MENU;
+import static com.watch.shop.app.controller.MenuConstants.START_SEPARATOR;
+import static com.watch.shop.app.controller.MenuConstants.TOTAL_COST_MESSAGE;
+
+@RequiredArgsConstructor
+public class MenuHandler {
+
+    private final WatchView watchView;
+    private final InputHandler inputHandler = new InputHandler();
+
+    public String printMainMenu() {
+        watchView.printMessage(MENU);
+
+        return inputHandler.readLineFromConsole();
+    }
+
+    public String printSortMenu() {
+        watchView.printMessage(SORT_MENU);
+
+        return inputHandler.readLineFromConsole();
+    }
+
+    public void displayAllWatches(List<Watch> watches) {
+        watchView.printMessage(START_SEPARATOR);
+        watchView.printFormattedMessage(FORMAT_HEADER, "Brand", "Model", "Color", "Type", "Price", "Arrival Date");
+        watchView.printFormattedMessage(FORMAT_SEPARATOR, "_".repeat(13), "_".repeat(27), "_".repeat(12), "_".repeat(12), "_".repeat(12), "_".repeat(14));
+
+        watches.forEach(w -> watchView.printFormattedMessage(FORMAT_ROW, w.getBrand(), w.getModel(), w.getColor(), w.getType(), w.getPrice(), w.getArrivalDate()));
+
+        watchView.printMessage(END_SEPARATOR);
+    }
+
+    public void displayTotalCost(BigDecimal totalCost) {
+        String message = TOTAL_COST_MESSAGE.formatted(totalCost);
+        watchView.printMessage(message);
+    }
+
+    public Watch printNewWatchMenu() {
+        watchView.printMessage("Enter Brand:");
+        String brand = inputHandler.readLineFromConsole();
+
+        watchView.printMessage("Enter Model:");
+        String model = inputHandler.readLineFromConsole();
+
+        watchView.printMessage("Enter Color (WHITE, SILVER, GOLD, BLUE, RED, BLACK):");
```

Comment:
Also I would advise not hardcode enum values.
Imagine we add new color to Color enum.
In such case you should seek through all application where you hardcoded colors in messages...

Just use something:

view.printMessage(String.format(ENTER_COLOR_MESSAGE, EnumSet.allOf(Color.class).toString());

and in Constants:
ENTER_COLOR_MESSAGE = "Enter Color %s:"

or even better, you could increase readibility and reduce duplication of code, since you have more enums (TYPE):
  view.printMessage(String.format(ENTER_COLOR_MESSAGE, extractEnumValues(Color.class));
  ...
   view.printMessage(String.format(ENTER_WATCH_TYPE_MESSAGE, extractEnumValues(WatchType.class));

    private <E extends Enum<E>> String extractEnumValues(Class<E> enumClass) {
        return EnumSet.allOf(enumClass).toString();
    }